### PR TITLE
Updated TBB search

### PR DESCRIPTION
**Pull request by mshabunin**
### This pullrequest changes

Tested with system packages in Ubuntu 16.04 and Fedora 25, official TBB distributions 4.1 - 4.4 (Linux) and 2017.0 (Linux/Windows/OSX), version built by _3rdparty/tbb_.

### CODE REVIEW

File: cmake/OpenCVDetectTBB.cmake [hunk 1]
```diff
@@ -1,94 +1,59 @@
...
+endfunction()
 
-if(NOT HAVE_TBB)
-  set(TBB_DEFAULT_INCLUDE_DIRS
-    "/opt/intel/tbb/include" "/usr/local/include" "/usr/include"
-    "C:/Program Files/Intel/TBB" "C:/Program Files (x86)/Intel/TBB"
-    "C:/Program Files/tbb/include"
-    "C:/Program Files (x86)/tbb/include"
-    "${CMAKE_INSTALL_PREFIX}/include")
+function(read_version _path)
```
Comment (alalek) [inline cmake/OpenCVDetectTBB.cmake]: Please avoid common names: `ocv_tbb_read_version`

**Comment by alalek** [comment]
:+1:

**Comment by alalek** [comment]
Could you take a look on this failed nightly build which blames TBB-related code: http://pullrequest.opencv.org/buildbot/builders/master_pack-android/builds/656/steps/build%20sdk/logs/stdio

**Comment by mshabunin** [comment]
Oh, looks like those compiler definitions were really needed - should be fixed by #7772

**Pull request merged**